src/ui/components/Rating/index.js

```diff
@@ -85,20 +94,35 @@ export class RatingBase extends React.Component<InternalProps> {
     );
   };
 
+  onHoverStar = (star: number) => {
+    if (this.props.readOnly) {
+      return;
+    }
+    this.setState({ hoveringOverStar: star });
+  };
+
+  stopHovering = () => {
+    this.setState({ hoveringOverStar: null });
+  };
+
   renderRatings() {
     const { readOnly } = this.props;
     // Accept falsey values as if they are zeroes.
     const rating = this.props.rating || 0;
 
     return [1, 2, 3, 4, 5].map((thisRating) => {
+      let isSelected = thisRating - rating <= 0.25;
+      if (this.state.hoveringOverStar !== null) {
+        isSelected = thisRating <= this.state.hoveringOverStar;
+      }
       const props = {
-        className: makeClassName('Rating-choice', {
-          'Rating-selected-star': thisRating - rating <= 0.25,
+        className: makeClassName('Rating-star', `Rating-rating-${thisRating}`, {
+          'Rating-selected-star': isSelected,
           'Rating-half-star':
             thisRating - rating > 0.25 && thisRating - rating <= 0.75,
         }),
-        id: `Rating-rating-${thisRating}`,
         key: `rating-${thisRating}`,
+        onMouseEnter: () => this.onHoverStar(thisRating),
```

Comment:
I couldn't get a pure CSS hover to work due to the `grid-gap` which causes a flicker. I couldn't think of an alternative that didn't involve implementing a custom gap which is a pitfall of doom. Switching to grid fixes so many other layout issues that I think it's an acceptable trade-off.